### PR TITLE
Maxi/scampersand reporting

### PR DESCRIPTION
**Pull request by melvinsoft**
The following PR introduces:
* A new playbook `postgresql_aggregation` that does the following:
    * Install psql client on the server.
    * Connects to the psql aggregation database and creates a new user and role for the current edxapp installation.
    * Creates a cron job to send the stats every six hours to the psql database.

* A small playbook to deploy this to our existing customers.

Please see the [related PR in edx-configs](https://github.com/noderabbit-team/edx-configs/pull/105)

### CODE REVIEW

**Comment by melvinsoft** [comment]
[(4) Install the Scampersand reporting app on Eucalyptus customer servers](https://trello.com/c/G31rBXND/1527-4-install-the-scampersand-reporting-app-on-eucalyptus-customer-servers)

**Comment by melvinsoft** [comment]
@tkeemon Really weird, I think it was an error doing the squashing and rebasing, it's fixed now, but you appear as the author of the commit, even I've reverted the changes for the un desired files.

File: playbooks/roles/postgresql_aggregation/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+
+- name: Install packages
+  apt:
+    name: "{{ item }}"
+    install_recommends: yes
+    state: present
+  with_items: postgresql_packages
```
Comment (mrgnr) [inline playbooks/roles/postgresql_aggregation/tasks/main.yml]: Add `tags: ['postgresql_aggregation']` to each task in this role.

File: playbooks/roles/postgresql_aggregation/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+- name: install requirements for role
+  pip:
+    name: "{{ item }}"
+    state: present
+  with_items: postgresql_aggregation_requirements
+
+- name: check if user already exists
```
Comment (mrgnr) [inline playbooks/roles/postgresql_aggregation/tasks/main.yml]: Is this task part of the workaround for https://github.com/ansible/ansible-modules-core/issues/297?

File: playbooks/appsemblerPlaybooks/eucalyptus_enterprise_deploy.yml [hunk 3]
```diff
@@ -112,3 +112,21 @@
...
+  hosts: "edxapp-server-0"
+  sudo: True
+  gather_facts: True
+  vars:
+    POSTGRESQL_AGGREGATION_INSTALL_CRON: true
+  roles:
+    - postgresql_aggregation
+
+- name: Install Scampersand Reporting
+  hosts: "edxapp-server-1"
```
Comment (melvinsoft) [inline playbooks/appsemblerPlaybooks/eucalyptus_enterprise_deploy.yml]: @tkeemon No, please see Aron comment on the slack channel: https://appsembler.slack.com/archives/C40PW2MU7/p1499391617740405

The cron job should run only in one of the edxapp servers.

File: playbooks/appsemblerPlaybooks/eucalyptus_pro_server.yml [hunk 4]
```diff
@@ -58,6 +59,7 @@
       when: "INSTALL_DEMO_DATA"
     - oauth_client_setup
     - forum
+    - postgresql_aggregation
```
Comment (tkeemon) [inline playbooks/appsemblerPlaybooks/eucalyptus_pro_server.yml]: Do we want to make this role conditional on `COMMON_ENVIRONMENT == "prod"` so this isn't installed on staging?

File: playbooks/roles/postgresql_aggregation/tasks/main.yml (hunk 2 above)
Comment (melvinsoft) [inline playbooks/roles/postgresql_aggregation/tasks/main.yml]: @mrgnr Yes, exactly!

File: playbooks/appsemblerPlaybooks/eucalyptus_enterprise_deploy.yml (hunk 3 above)
Comment (mrgnr) [inline playbooks/appsemblerPlaybooks/eucalyptus_enterprise_deploy.yml]: Sure, it's fine for this particular thing :) In general, it becomes harder to manage servers the more differences we have.

**Comment by melvinsoft** [comment]
@tkeemon @mrgnr Thanks for reviewing this, all the requesting changes have been addressed, please take another look.

**Pull request merged**